example/mismatched/input.go

```diff
@@ -0,0 +1,58 @@
+//go:generate go run github.com/jmattheis/goverter/cmd/goverter github.com/jmattheis/goverter/example/mismatched
+package mismatched
+
+/*
+This example demonstrates how to handle structures that have mismatched
+pointers, that for historical reasons have to be maintained as-is.
+
+In this example:
+- DBCustomer is DBPerson, *DBAddress
+- APICustomer is *APIPerson, APIAddress
+*/
+
+// goverter:converter
+// goverter:extend ToAPICustomer
+type Converter interface {
+	Convert(customers DBCustomers) APICustomers
+	ToAPIPerson(DBPerson) *APIPerson
+	ToAPIAddress(DBAddress) APIAddress
+}
+
+func ToAPICustomer(c Converter, customer DBCustomer) APICustomer {
+	var result APICustomer
+	result.APIPerson = c.ToAPIPerson(customer.DBPerson)
+	if customer.DBAddress != nil {
+		result.APIAddress = c.ToAPIAddress(*customer.DBAddress)
+	}
+	return result
+}
```

Comment:
In this example, you implement a custom converter of DBCustomer to APICustomer, this isn't strictly needed (still perfectly fine). You can only define a custom converter for `*DBAddress` to `APIAddress` because this is the only conversion that goverter cannot infer. It could look like this:

```golang
// goverter:converter
// goverter:extend AddressOrDefault
type Converter interface {
	Convert(customers DBCustomers) APICustomers

	// goverter:map DBPerson APIPerson
	// goverter:map DBAddress APIAddress
	ToApiCustomer(DBCustomer) APICustomer

	ToAPIAddress(DBAddress) APIAddress
}

func AddressOrDefault(c Converter, address *DBAddress) APIAddress {
	if address == nil {
		// use APIAddress with default values if DBAddress is unset
		return APIAddress{}
	}
	return c.ToAPIAddress(*address)
}
```
What do you think? I'd be okay with merging it as is.